### PR TITLE
feat(cockpit): paginate replay endpoint for large sessions

### LINKED ISSUE #1705: Cockpit daemon: replay pagination / NDJSON streaming for large sessions
**Note:** Claude handled the investigation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

Split out from the cockpit-in-TUI follow-ups epic #1113.

## Problem

The cockpit replay endpoint returns the full event history in a single JSON array. For a long session this buffers every event in daemon memory and ships one large response. The epic scoped this as "only matters if profiling shows large-session lag", so this issue tracks the optimization and is deliberately gated on evidence.

## What's there today

- `src/server/api/cockpit.rs:1229` (`cockpit_replay`) loads all events via `replay_from`, maps them to `CockpitBroadcastFrame`s, and returns a single `Json(ReplayResponse { frames, lost, highest_seq, lowest_seq })`.
- `src/cockpit/event_store.rs:259` (`replay_from`) runs `SELECT seq, event_json ... WHERE seq > ?2 ORDER BY seq ASC` and collects every match into a `Vec<(u64, Event)>`.

So both the daemon memory footprint and the response size scale with total events since `since`.

## Proposal

Gated on profiling. Option A: add `limit` / `cursor` query params to `cockpit_replay` and `replay_from`, returning `next_cursor` / `has_more` for cursor-based pagination. Option B: stream NDJSON (chunked transfer, one frame per line) so neither side buffers the whole history. Either path requires the consumers (`web/src/hooks/useCockpit.ts` replay handling and the TUI replay path) to read incrementally.

Touch points: `src/server/api/cockpit.rs`, `src/cockpit/event_store.rs`, plus the web and TUI replay consumers.

## Why now

- This is low priority and explicitly speculative. There is no profiling data yet showing large-session lag. Filed to track the optimization, not to schedule it ahead of a measured need.

## User stories / regression tests

Implementation MUST add tests covering each story below; PR is not done without them.

- As a cockpit user with a long session, when I attach, then replay loads incrementally or streams rather than blocking on one large array, and the rebuilt transcript matches the single-array result.
  - Suggested test layer: Rust integration test against a seeded large session, plus `web/tests/live/` for the consumer.

## Open questions

- Pagination vs NDJSON streaming, which fits the existing reducer better?
- What event count threshold actually warrants this? Needs a profiling number first.
- Is this premature until profiling shows a problem?

## Out of scope

- Event-store schema changes.
- The live WebSocket broadcast path (this is replay only).

## Cross-references

- Parent epic: #1113.

### PR DESCRIPTION
**Pull request by Seluj78**
**Note:** Claude handled the implementation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

## Description

The cockpit replay endpoint `GET /api/sessions/{id}/cockpit/replay?since=N` loaded the entire event history (`seq > since`) into a `Vec` and shipped it as one JSON array, so both daemon memory and response size scaled with total events in a session. This adds keyset pagination over `seq` so neither side buffers the whole history at once.

`EventStore::replay_page(since, limit)` probes `LIMIT n + 1` rows, so `has_more` is computed under the same lock as the page rows (it never races a concurrently growing `highest_seq`), and it reports `last_scanned_seq` as the next cursor. The cursor advances over rows that fail to deserialise, so a corrupt row can never trap a paging loop. `replay_from` now delegates to it with no limit and keeps backing the WS on-connect drain unchanged.

`ReplayQuery` gains `limit`; `ReplayResponse` gains `next_cursor` and `has_more` (all `#[serde(default)]`). Omitted `limit` now falls back to a bounded default page (1000, hard max 2000) rather than unbounded, so a naive client cannot make the daemon buffer a whole long session. This is an intentional behavior change to the replay endpoint; cockpit client and daemon ship in the same binary, and all three in-repo consumers (web, TUI, `cockpit history`) were updated to page via `has_more`/`next_cursor`. The metadata-only status probe keeps the unbounded `replay`.

Both client loops are capped at the first page's `highest_seq`: events appended after replay began arrive over the live WS channel and are deduped by the reducer, so chasing them in the replay loop would never converge on a busy session. `lost` is honored on every page (a retention prune between pages can open a real gap after page one), not only the first.

Design chosen (cursor pagination over NDJSON streaming) because both reducers already read incrementally and dedupe by seq, whereas NDJSON would force a chunked-body rewrite in three consumers and hold the SQLite mutex across an async boundary.

Closes #1705

## PR Type

- [x] New Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Documentation
- [ ] Infrastructure / CI

## How to test

- [ ] On a cockpit session with a long transcript, attach via the web dashboard and confirm the full transcript loads with no missing events (the Network tab shows several bounded `/cockpit/replay?...&limit=1000` requests instead of one large response).
- [ ] `aoe cockpit history <session>` on a long session prints the complete event list (now fetched in pages under the hood).
- [ ] Open the cockpit view in the TUI on a long session; confirm the historical transcript renders fully on attach and again after a Lagged reset.
- [ ] `curl '.../api/sessions/<id>/cockpit/replay?since=0&limit=10'` returns at most 10 frames plus `next_cursor` and `has_more: true`; passing `next_cursor` back as `since` and repeating reassembles the whole history, and the final page reports `has_more: false`.

## Checklist

- [x] New and existing tests pass
- [x] Documentation was updated where necessary
- [ ] For UI changes: included screenshot or recording

## Test Coverage Analysis

**Web dashboard / cockpit (Playwright user story)**
- [ ] N/A: this PR doesn't change a user-facing dashboard flow
- [x] Added or updated a Playwright spec under `web/tests/` and updated `web/tests/coverage-matrix.json`
- [ ] Skipped a test, because:

**TUI / CLI (e2e test)**
- [ ] N/A: this PR doesn't change TUI rendering, a CLI subcommand, or session lifecycle
- [ ] Added or updated an e2e test under `tests/e2e/`
- [x] Skipped a test, because: the TUI/CLI behavior is unchanged (replay just pages internally now); the paging path they share is covered by Rust store + client unit tests and a live Playwright pagination assertion against the same endpoint.

## AI Usage

- [ ] No AI was used
- [x] AI was used

**AI Model/Tool used:** Claude (via Claude Code, agent-of-empires `/aoe-implement` skill)

**Any Additional AI Details you'd like to share:**
Investigation, plan, and implementation drafted by Claude under my direction, with a multi-model `/debate` pass that hardened the cursor/`has_more` correctness and the default-bound decision. I reviewed each commit, made the design calls, and ran the validation.

**NOTE:**
When responding to reviewer questions, please respond yourself rather than copy/pasting reviewer comments into an AI and pasting back its answer. We want to discuss with you, not your AI :)

- [x] I am an AI Agent filling out this form (check box if true)

## Overview

This PR changes the cockpit replay flow from returning an entire session's event history as one large JSON array to cursor-based, paginated replay. Replays are delivered in bounded pages so clients and the daemon avoid unbounded memory and response growth for long sessions.

## What changed (non-technical)

- Server now returns replay results in pages instead of one full array.
- Responses include pagination metadata so clients can request the next page.
- Web, TUI, and CLI clients updated to fetch replay pages incrementally.
- Server enforces a bounded default page size and a hard maximum to prevent unbounded buffering.
- Unbounded replay is preserved for metadata/status probes only.

## Benefits

- Prevents unbounded memory and network usage when replaying long sessions.
- Faster first-page responsiveness for clients.
- More robust replay: pagination logic skips corrupt rows to avoid paging loops.
- Live events continue arriving via WebSocket and are deduplicated by existing reducers.

## Technical details (brief)

- New API fields: ReplayQuery.limit: Option<u64>; ReplayResponse.next_cursor: Option<u64> and has_more: bool (serde-defaulted for back-compat).
- EventStore::replay_page(session_id, since, limit) added; returns ReplayPage { events, last_scanned_seq, has_more, highest_seq, lowest_seq }.
- replay_page uses keyset pagination (seq > since), probes limit+1 rows to compute has_more, and advances last_scanned_seq past corrupt rows so pagination resumes correctly.
- replay_from delegates to replay_page with no limit to preserve legacy unbounded behavior for probes.
- Server clamps requested limits to a default (1000) and hard max (2000).
- Clients: added replay_page and replay_paged helpers and a REPLAY_PAGE_SIZE constant; clients iterate pages using next_cursor/has_more and stop at the first page’s highest_seq so concurrent appends are handled via live WS.
- Tests: serde back-compat tests for missing pagination fields, and replay tests ensuring paginated sequence reconstruction matches unpaginated output, correct boundary handling, and corrupt-row skipping.

### CODE REVIEW

File: src/cockpit/event_store.rs [hunk 1]
```diff
@@ -323,54 +341,116 @@ impl EventStore {
...
-        let rows = match stmt.query_map(params![session_id, since as i64], |row| {
+        let map_row = |row: &rusqlite::Row| {
             let seq: i64 = row.get(0)?;
             let json: String = row.get(1)?;
             Ok((seq as u64, json))
-        }) {
+        };
+        let rows = match probe {
+            Some(p) => stmt.query_map(params![session_id, since as i64, p as i64], map_row),
+            None => stmt.query_map(params![session_id, since as i64], map_row),
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply src/cockpit/event_store.rs]: Good catch, real bug. `since=u64::MAX` from the status probe cast to `-1`, so `seq > -1` returned the whole transcript. Clamped with `i64::try_from(since).unwrap_or(i64::MAX)` before binding, plus a `replay_page_since_u64_max_returns_empty` regression test. Fixed in https://github.com/agent-of-empires/agent-of-empires/pull/1753/commits/2cf70b5b44a03708933a833a2c95ad79cae0f643

File: src/server/api/cockpit.rs [hunk 2]
```diff
@@ -1664,8 +1671,23 @@ pub async fn cockpit_replay(
...
+    let limit = q
+        .limit
+        .map(|l| l as usize)
+        .unwrap_or(DEFAULT_REPLAY_PAGE)
+        .clamp(1, MAX_REPLAY_PAGE);
+    let page = state
+        .cockpit_event_store
+        .replay_page(&id, q.since, Some(limit));
+    let next_cursor = page.last_scanned_seq;
+    let has_more = page.has_more;
```
[the remark this answers is not in the document]
Comment (Seluj78) [reply src/server/api/cockpit.rs]: Agreed, the same race a pre-merge design review flagged. `replay_page` now reads `highest_seq`/`lowest_seq` under the same lock as the page rows and returns them on `ReplayPage`; the handler dropped its two separate store calls, so a concurrent `record()` can no longer desync the cap from `has_more`. Fixed in https://github.com/agent-of-empires/agent-of-empires/pull/1753/commits/2cf70b5b44a03708933a833a2c95ad79cae0f643

**Comment by Seluj78** [comment]
**Note:** This comment was written by Claude (via Claude Code) on my behalf. — @Seluj78

> 🧹 Nitpick (src/cockpit/protocol.rs): Add a serde back-compat regression test for the new paging fields.

Added `replay_query_defaults_limit_when_absent` and `replay_response_defaults_paging_fields_when_absent` in the protocol `#[cfg(test)]` module, asserting older payloads without `limit`/`next_cursor`/`has_more` still deserialize. Done in https://github.com/agent-of-empires/agent-of-empires/pull/1753/commits/49a52c74b90a4586d761d5548330da6c8469bfcc
